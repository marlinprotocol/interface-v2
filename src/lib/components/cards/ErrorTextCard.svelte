<script lang="ts">
	import InputCard from '$lib/atoms/v2/cards/InputCard.svelte';
	import Text from '$lib/atoms/v2/texts/Text.svelte';

	export let showError = false;
	export let errorMessage = 'Error Message';
	export let styleClass = 'mt-4';
</script>

{#if showError}
	<InputCard variant="warning" {styleClass}>
		<Text variant="small" styleClass="text-red-500 my-2 whitespace-normal" text={errorMessage} />
	</InputCard>
{/if}
