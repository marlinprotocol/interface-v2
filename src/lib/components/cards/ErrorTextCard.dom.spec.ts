import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/svelte';
import ErrorTextCard from './ErrorTextCard.svelte';

describe('ErrorTextCard', () => {
	it('does not render when showError is false', () => {
		const { container } = render(ErrorTextCard, {
			props: {
				showError: false,
				errorMessage: 'This is an error!',
				styleClass: 'mt-4'
			}
		});

		expect(container).toMatchSnapshot();
	});

	it('renders correctly with an error message when showError is true', () => {
		const { container } = render(ErrorTextCard, {
			props: {
				showError: true,
				errorMessage: 'This is an error!',
				styleClass: 'mt-4'
			}
		});

		expect(container).toMatchSnapshot();
	});

	it('updates the snapshot when the error message changes', () => {
		const { container } = render(ErrorTextCard, {
			props: {
				showError: true,
				errorMessage: 'A different error message!',
				styleClass: 'mt-4'
			}
		});

		expect(container).toMatchSnapshot();
	});
});
