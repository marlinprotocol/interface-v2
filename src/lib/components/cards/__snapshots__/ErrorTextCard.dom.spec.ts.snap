// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorTextCard > does not render when showError is false 1`] = `
<body>
  <div>
    
    <!--&lt;ErrorTextCard&gt;-->
  </div>
</body>
`;

exports[`ErrorTextCard > renders correctly with an error message when showError is true 1`] = `
<body>
  <div>
    <div
      class="w-full rounded-lg bg-red-100 px-4 py-1 mt-4"
      data-testid="input-card"
      id=""
    >
      <div
        class="text-red-500 my-2 whitespace-normal text-sm font-normal"
        data-testid="text"
      >
        This is an error!
      </div>
      <!--&lt;Text&gt;-->
    </div>
    <!--&lt;InputCard&gt;-->
    
    <!--&lt;ErrorTextCard&gt;-->
  </div>
</body>
`;

exports[`ErrorTextCard > updates the snapshot when the error message changes 1`] = `
<body>
  <div>
    <div
      class="w-full rounded-lg bg-red-100 px-4 py-1 mt-4"
      data-testid="input-card"
      id=""
    >
      <div
        class="text-red-500 my-2 whitespace-normal text-sm font-normal"
        data-testid="text"
      >
        A different error message!
      </div>
      <!--&lt;Text&gt;-->
    </div>
    <!--&lt;InputCard&gt;-->
    
    <!--&lt;ErrorTextCard&gt;-->
  </div>
</body>
`;
